Fix formatting for super and self keywords

DaelonSuzuka opened the pull request:
Fixes #890

DaelonSuzuka committed:
Disable very verbose test logging

DaelonSuzuka committed:
Fix super and self keywords

theXYZT commented:
I may be reading this wrong, but it doesn't seem like this fixes: `[self] -> [ self ]`.

DaelonSuzuka commented:
No it doesn't. (when would that even happen?)

theXYZT commented:
> No it doesn't. (when would that even happen?)

When you use 'self' as a key for a Dictionary.

DaelonSuzuka commented:
Yes that's obvious but I'm struggling to imagine when I would ever want that.

DaelonSuzuka committed:
Fix extra spaces around self

DaelonSuzuka commented:
@theXYZT can you think of any other cases?

<img width="285" height="218" alt="image" src="https://github.com/user-attachments/assets/28648546-8fcf-4c68-ad88-b3edca77c29b" />

theXYZT commented:
In your image, is `{ self: self }` supposed to be the formatted text?

DaelonSuzuka commented:
> In your image, is { self: self } supposed to be the formatted text?

Hm, I just doubled checked and other dict literals are not expanded here. I've been writing mostly ts lately and a similar object literal does have the extra padding.

Just checked python/ruff, and padding is not added to dict literals there.

I have intentionally avoided adding more rules for this syntax because my formatter does not modify indentation at all. A proper formatter would re-indent list/dict literals, collapse small ones onto one line, expand large ones onto multiple lines, etc. Mine is not smart enough to handle that so I've just left them alone.

Let me mess with it a bit more...

DaelonSuzuka committed:
Fix type error

DaelonSuzuka committed:
Remove old snapshot template folder

DaelonSuzuka committed:
Implement super/self formatting rules a different way

DaelonSuzuka commented:
<img width="313" height="228" alt="image" src="https://github.com/user-attachments/assets/307622f3-b1fe-4eac-bb44-cc00e4a96e90" />

There we go, much better. I removed the special cases entirely and changed it to treat self/super like regular identifiers. Thanks!

DaelonSuzuka committed:
Add another test case

Calinou commented on src/debugger/godot4/variables/debugger_variables.test.ts:
The lines being changed here should probably be removed entirely rather than commented out (since you can find them back using version control).

theXYZT commented:
Is there anything blocking this from being pushed? I'd love to have a nightly build with this.

Calinou commented:
@DaelonSuzuka Is this good to merge now? It appears this has regressed in 2.6.0: https://github.com/godotengine/godot-vscode-plugin/issues/965

Calinou reviewed (approved):
Thanks!
